ports/jniapi/src/lib.rs

```diff
@@ -873,6 +873,10 @@ fn get_options(
     prefs.insert("js.baseline_jit.enabled".to_string(), false.into());
     prefs.insert("js.ion.enabled".to_string(), false.into());
 
+    if let Some(url) = url {
+        prefs.insert("shell.homepage".to_string(), url.into());
```

Comment:
Overriding the homepage preference to pass along the URL to load doesn't seem like the best approach here.
For example, if we always override the preference here, we will not be able to support a 'go to homepage' button in the UI to navigate away from the loaded URL.

We should probably just extend `InitOptions` to take an optional `url` member and handle it in [simpleservo::init](https://github.com/servo/servo/blob/66563ed0273c05ab9d687c883649521d14d610e8/ports/jniapi/src/simpleservo.rs#L259).

For reference, `servoshell` also has similar [fallback logic](https://github.com/servo/servo/blob/66563ed0273c05ab9d687c883649521d14d610e8/ports/servoshell/app.rs#L87) and there were some internal discussions about refactoring out the logic common to both ports, but that will be a bigger effort, so for now handling it in simpleservo::init should be fine.

Reply:
Should be fine now